### DISCUSSION TITLE
Option to add custom command before a target

### DISCUSSION THREAD

**Discussion by ianisl** [visose/Robots#150] [Q&A] [answered]
This is not an issue but a suggestion/question: 

Would it be possible to add an option to the Create target component in order to decide whether the code provided via the Command input is added before or after the target's motion instruction? 

I believe this could improve the GH code legibility for some complex scenarios.

**Comment by visose** [comment, marked as answer]
You can use the Discussions section for questions and suggestions.

You can do this via a scripting component. 

It is not exposed in Grasshopper to keep things simple. People get overwhelmed if there are too many options and might not know when to use them. As it is not completely necessary (you can always add the command on the previous target) the trade-off of keeping things as simple as possible won the argument here. There are multiple other similar things only exposed through scripting.

**Reply by ianisl** [reply]
Sure, this will add some unnecessary complexity and I understand you'd prefer to avoid feature creep.
Adding the command on the previous target is a solution but sometimes not the most practical or legible thing. But yes this is easy to do with scripting. Thanks for the reply!

**Reply by visose** [reply]
For example:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/5129614/157655012-7e6c3e3d-58dd-4827-8c19-5b601e2ccb72.png">

```
  private void RunScript(object x, object y, ref object A)
  {
    var command = (Robots.Command) x;
    command.RunBefore = true;
    A = command;
  }
```
You need to add an assembly reference to Robots.dll, located in `C:\Users\{user}\AppData\Roaming\McNeel\Rhinoceros\packages\7.0\Robots\{latest version}`

This method modifies the original command (it doesn't duplicate the command, its still the same one), so avoid using the output of the original command component elsewhere (it would be safer to duplicate the command before modifying it).

**Reply by ianisl** [reply]
Thanks a lot for the example!
I was about to edit my previous message since I just realized you meant scripting with reference to `Robots.dll`. I had first understood it as a suggestion to post-process the code via regexes or something similar (which is what I'm currently doing).
